app/src/test/java/org/gdg/frisbee/android/chapter/ChapterComparatorTest.java

```diff
@@ -0,0 +1,117 @@
+    private static final Chapter chapterIstanbul = new Chapter("GDG Istanbul", "100514812580249787371");
+    private static final Chapter chapterWithDoubleSpaceInName = new Chapter("GDG  Mukono", "108217555392256442794");
+
+    @Before
+    public void setup() {
+        locationIstanbul = new Location(LocationManager.GPS_PROVIDER);
+        locationIstanbul.setLatitude(28.97702);
+        locationIstanbul.setLongitude(41.011511);
+
+        chapterStartsWithLetterA.setGeo(new Geo(7.083333, 4.833333));
+        chapterStartsWithDigit.setGeo(new Geo(29.938126, 30.91398));
+        chapterIstanbul.setGeo(new Geo(28.97696, 41.00527));
+    }
+
+    @Test
+    public void shouldBeAlphabeticalWhenNoHomeAndLocation() {
+        ChapterComparator comparator = new ChapterComparator(null, null);
+
+        assertElementsOrderedLikeThis(comparator,
+                chapterStartsWithDigit,
+                chapterStartsWithLetterA,
+                chapterIstanbul);
+    }
+
+    @Test
+    public void doubleSpaceInNameShouldNotBeFirst() {
+        ChapterComparator comparator = new ChapterComparator(null, null);
+
+        assertElementsOrderedLikeThis(comparator,
+                chapterStartsWithDigit,
+                chapterStartsWithLetterA,
+                chapterWithDoubleSpaceInName);
+    }
+
+    @Test
+    public void nearestLocationShouldBeTop() {
+        ChapterComparator comparator = new ChapterComparator(null, locationIstanbul);
+
+        assertElementsOrderedLikeThis(comparator,
+                chapterIstanbul,
+                chapterStartsWithDigit,
+                chapterStartsWithLetterA);
+    }
+
+    @Test
+    public void chapterWithoutLocationShouldBeSortedAlphabetically() {
+        ChapterComparator comparator = new ChapterComparator(null, locationIstanbul);
+
+        assertElementsOrderedLikeThis(comparator,
+                chapterIstanbul,
+                chapterStartsWithDigit,
+                chapterWithoutLocation);
+    }
+
+    @Test
+    public void homeChapterShouldBeTop() {
+        ChapterComparator comparator =
+                new ChapterComparator(chapterIstanbul.getGplusId(), locationIstanbul);
+
+        assertElementsOrderedLikeThis(comparator,
+                chapterIstanbul, chapterWithoutLocation);
+        assertElementsOrderedLikeThis(comparator,
+                chapterIstanbul, chapterStartsWithDigit);
+        assertElementsOrderedLikeThis(comparator,
+                chapterIstanbul, chapterStartsWithLetterA);
+        assertElementsOrderedLikeThis(comparator,
+                chapterIstanbul, chapterWithDoubleSpaceInName);
+    }
+
+    private <T> void assertElementsOrderedLikeThis(
+            Comparator<? super T> comparator, T... elements) {
+        List<T> expectedOrder = Arrays.asList(elements);
+
+        List<T> shuffledAndSorted = new ArrayList<>(expectedOrder);
+        Collections.shuffle(shuffledAndSorted, new Random(0));
```

Comment:
I got the code from @swankjesse and didn't change it. Here is the link. 
https://publicobject.com/2007/04/helper-method-for-unit-testing.html

But you are right. For our case, it doesn't add much and it may even cause problems.